sys/cddl/contrib/opensolaris/uts/common/sys/dtrace.h

```diff
@@ -1323,14 +1323,27 @@ typedef struct {
 							/* start tracing */
 #define	DTRACEIOC_STOP		_IOWR('x',13,processorid_t)
 							/* stop tracing */
-#define	DTRACEIOC_AGGDESC	_IOW('x',15,dtrace_aggdesc_t *__kerncap)
+#define	DTRACEIOC_AGGDESC	_IOW('x',15,dtrace_aggdesc_t *)
 							/* get agg. desc. */
 #define	DTRACEIOC_FORMAT	_IOWR('x',16,dtrace_fmtdesc_t)	
 							/* get format str */
-#define	DTRACEIOC_DOFGET	_IOW('x',17,dof_hdr_t *__kerncap)
+#define	DTRACEIOC_DOFGET	_IOW('x',17,dof_hdr_t *)
 							/* get DOF */
 #define	DTRACEIOC_REPLICATE	_IOW('x',18,dtrace_repldesc_t)	
 							/* replicate enab */
+
+#if __has_feature(capabilities)
+typedef struct {
+	void *__capability dof;		/* DOF userland address written to driver. */
+	int	n_matched;	/* # matches returned by driver. */
+} dtrace_enable_io_t_cap;
```

Comment:
I would generally prefer that you kept the __kerncap changes and added a dtrace_enable_io64_t for the old ones as that's the current model.